src/Bicep.LangServer/Snippets/SnippetsProvider.cs

```diff
@@ -135,37 +141,51 @@ private void CacheResourceDeclarationAndDependencies(string template, string man
                 {
                     if (declaredSymbol.Type is TypeSymbol typeSymbol && typeSymbol.TypeKind != TypeKind.Error)
                     {
-                        string type = typeSymbol.Name;
-                        CacheResourceDeclaration(resourceDeclarationSyntax, type, template, description);
-                        CacheResourceDependencies(kvp.Value, template, type);
+                        CacheResourceDeclaration(resourceDeclarationSyntax, typeSymbol, template, description, manifestResourceName);
+                        CacheResourceDependencies(typeSymbol, kvp.Value, template);
                     }
                 }
             }
         }
 
-        private void CacheResourceDeclaration(ResourceDeclarationSyntax resourceDeclarationSyntax, string type, string template, string description)
+        private void CacheResourceDeclaration(ResourceDeclarationSyntax resourceDeclarationSyntax, TypeSymbol typeSymbol, string template, string description, string manifestResourceName)
         {
-            if (!resourceTypeToBodyMap.ContainsKey(type))
+            string type = typeSymbol.Name;
+
+            if (!resourceTypeInfoMap.ContainsKey(type))
```

Comment:
>type [](http://example.com/codeflow?start=49&length=4)

Shouldn't you also update this to use ResourceTypeReference as key? Type names are not guaranteed to be unique in our type system.

Reply:
Updated to use and cache ResourceTypeReference instead of type name everywhere as part of this commit:
https://github.com/Azure/bicep/commits/dev/bhsubra/ShowChildSnippets